src/Build/BackEnd/BuildManager/BuildParameters.cs

```diff
@@ -394,7 +394,7 @@ public bool ResetCaches
         public bool EnableNodeReuse
         {
             get => _enableNodeReuse;
-            set => _enableNodeReuse = value;
+            set => _enableNodeReuse = Environment.GetEnvironmentVariable("MSBUILDDISABLENODEREUSE") == "1" ? false : value;
```

Comment:
Please move `MSBUILDDISABLENODEREUSE` to `shared/traits.cs` and reference it from there, that way we don't have to do an actual environment variable lookup in this setter.